java-manual/modules/ROOT/pages/value-mapping.adoc

```diff
@@ -0,0 +1,173 @@
+var persons = driver.executableQuery("MERGE (p:Person {name: 'Margarida', age: 29}) RETURN p")
+    .withConfig(QueryConfig.builder().withDatabase("neo4j").build())
+    .execute()
+    .records()
+    .stream()
+    .map(record -> record.get("p").as(Person.class))
+    .toList();
+System.out.println(persons.get(0));  // Person[firstName=Margarida, Years=29]
+----
+
+
+== Map driver records to a local class
+
+The earlier examples have mapped a driver value (for example a node identified with `p`) to a class.
+You can also use the mapping feature with driver records, through the link:https://neo4j.com/docs/api/java-driver/current/org.neo4j.driver/org/neo4j/driver/Record.html#as(java.lang.Class)[`Record.as()`] method.
+
+.Return keys `name`/`p.age` are mapped to the object attributes `Name`/`Age`
+[source, java]
+----
+// import org.neo4j.driver.mapping.Property;
+
+record Person(String name, @Property("p.age") Integer age) {}
+var persons = driver.executableQuery("""
+            MERGE (p:Person {name: 'Margarida', age: 29})
+            RETURN p.name AS name, p.age
+        """)
+    .withConfig(QueryConfig.builder().withDatabase("neo4j").build())
+    .execute()
+    .records()
+    .stream()
+    .map(record -> record.as(Person.class))
+    .toList();
+System.out.println(persons.get(0));  // Person[name=Margarida, age=29]
+----
+
+
+[#constructors]
+== Work with multiple constructors
+
+Your Java record class can contain multiple constructors.
+In that case, the driver picks one basing on the following criteria (in order of priority):
+
+- Most matching properties
+- Least mis-matching properties
+
+At least one property must match for a constructor to work with the mapping.
+
+.An additional constructor to handle the optional `age` property
+[source, java]
+----
+// import org.neo4j.driver.mapping.Property;
+
+record Person(String name, int age) {
+    public Person(@Property("name") String name) {
+        this(name, -1);
+    }
+}
+var persons = driver.executableQuery("MERGE (p:Person {name: 'Axel'}) RETURN p")
+    .withConfig(QueryConfig.builder().withDatabase("neo4j").build())
+    .execute()
+    .records()
+    .stream()
+    .map(record -> record.get("p").as(Person.class))
+    .toList();
+----
+
+[NOTE]
+====
+The compiler renames constructor parameters by default, unless the compiler `-parameters` option is used or the parameters belong to the cannonical constructor of `java.lang.Record`.
+
+In the example above, the constructor containing only `name` uses the `@Property` annotation even if it doesn't specify a different name than the constructor argument. This is needed because that is not the canonical constructor.
+====
+
+
+== Insert and update data
+
+You can also use the mapping feature to insert or update data, by creating an instance of the `record` object that serves as a blueprint for your object and then passing it to the query as a parameter.
```

Comment:
I think this should emphasise that it works for Java `record` only. No support is planned for standard classes at the moment.

For this reason, I think we should recommend using Java `record` with Value Mapping feature, see line `14`.